src/gurt/telem/telem_producer.c

```diff
@@ -0,0 +1,292 @@
+/*
+ * (C) Copyright 2020 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. 8F-30005.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+/*
+ * This file shows an example of using the telemetry API to produce metrics
+ */
+
+#include <stdio.h>
+#include "gurt/common.h"
+#include "gurt/telemetry.h"
+
+/*
+ * A sample function that creates and incremements a metric for a loop counter
+ */
+void test_function1(int count)
+{
+	static struct d_tm_node_t	*loop = NULL;
+	int	rc;
+	int 	i;
+
+	for (i = 0; i < count-1; i++) {
+		rc = d_tm_increment_counter(&loop, __FILE__, __FUNCTION__,
+					    "loop counter", NULL);
+		if (rc != 0) {
+			return;
+		}
+	}
+
+	/*
+	 * Demonstrates how the metric is accessed by the initialized pointer.
+	 * No name is required.  The API uses the initialized pointer if it is
+	 * provided, and only uses the name if the pointer doesn't reference
+	 * anything.
+	 */
+	rc = d_tm_increment_counter(&loop, NULL);
+
+
+}
+
+/*
+ * A sample function that creates and records a timestamp when this function is
+ * called.
+ */
+void test_function2()
```

Comment:
(style)  Bad function definition - void test_function2() should probably be void test_function2(void)